vividus-plugin-web-app-to-rest-api/src/main/java/org/vividus/crawler/CrawlControllerFactory.java

```diff
@@ -97,7 +101,28 @@ private CrawlConfig createCrawlConfig(URI mainApplicationPage) throws IOExceptio
     }
 
     @JsonIgnoreProperties(ignoreUnknown = true)
+    @JsonAutoDetect(fieldVisibility = Visibility.NONE)
+    @SuppressWarnings("unused")
     private static final class MappedCrawlConfig extends CrawlConfig
     {
+        private Http http;
+
+        public void setHttp(Http http)
+        {
+            this.http = http;
+        }
+
+        @SuppressWarnings({ "EmptyPublicCtorInClass", "RedundantModifier" })
+        private final class Http
+        {
+            public Http()
```

Comment:
why is it needed?

Reply:
required by jackson: non-static inner classes like this can only by instantiated using default, no-argument constructor

Reply:
is package-private an option here?

Reply:
no, it has to be public